### DISCUSSION TITLE
What is the difference between version 10 and 9?

### DISCUSSION THREAD

**Discussion by mohamad-asghari** [dnnsoftware/Dnn.Platform#5782] [General] [answered]
What capabilities and features are going to be released in version 10?

**Comment by valadas** [comment, marked as answer]
The main purpose of v10 is to remove any APIs that were marked as as deprecated on Dnn v8 and before. It is also an opportunity to update some dependencies that have a bigger risk of being braking changes for 3rd party extensions.

Then comes a big wishlist, keep in mind that DNN is open-source and any items on the list depend on someone contributing the feature, so no guarantees on any of them.
However the following is done:
- Dependency injection is available in way more places.
- CodeDOM will be installed by default to make more recent versions of c# available in Razor contexts.
- Support for API Tokens.

And not complete yet but we would love to have this as part of v10:
- A new default theme that is dependency-free
- Site Settings for CSS Variables with an editor (to allow providing simple things like branding colors etc. to any 3rd party that needs to know about those). (make themes more reusable, harmony with module styles, more branded persona bar, etc.) (this one was done a long time ago by me but never delivered because we decided to focus on stabilizing v9 and removing Telerik instead of focusing on new features. We are at a nice spot now with v9 to start moving forward...
- Integrating Polydeploy (which will be named something more generic) into the core for deployment of extensions.

I may be forgetting some other things here...

But the main thing to know is that v10.0.0 is where we need to make those breaking changes. Anything else could very well come in a v10.1.0 or v10.2.0, etc. But some of those things may get into 10.0.0 if they are contributed early enough.

**Reply by bdukes** [reply]
One other breaking change for version 10 is the forced removal of the old, obsolete Telerik component upon upgrading (new installations of DNN already do not include it).

**Comment by GerardSmit** [comment]
You can use the following query in pull requests to get all the DNN 10-related pull requests that are merged:

```
is:pr is:merged milestone:10.0.0 NOT in:title develop NOT in:title 9.x.x 
```

https://github.com/dnnsoftware/Dnn.Platform/pulls?q=is%3Apr+is%3Amerged+milestone%3A10.0.0+NOT+in%3Atitle+develop+NOT+in%3Atitle+9.x.x

I've excluded the words "develop" and "9.x.x" because these are PRs to bring changes from DNN 9 to DNN 10.
